### ISSUE TITLE
regarding boundary condtions in Example1 1D wave propagation problem.

### ISSUE THREAD

**Issue by Abhishekc95** [hl4220/Finite_PINN#2]
Thank you for sharing the code for Finite PINN paper.
Have you tried for 1D problem to use boundary condtions?. In this case you are using  x_bc, t_tbc, y_tbc for out_tbc(training for a given wavefield at x =0 from .mat file) to solve wave reflection problem. and IC is satisfed by hard constraint(t^2), isnt this an ill posed problem? as the boundary condition on the other is not specified?
have you tried using free condition or fixed condition on the other end to solve the problem(bc loss) along with pde loss?
edit: i realised the LBO basis function satisfies the neumann boundary condition on the other end.

**Comment by hl4220** [comment]
Hi there. Thanks for your comments. Like what you said the free Neumann boundary conditions are automatically satisfied at the free end.

**Comment by Abhishekc95** [comment]
here you used 5 cycle hanning window function. have you tried with narrow banded pulse? i mean shorter wavelength than one you have used?

**Comment by hl4220** [comment]
I didn't try that but I think it makes little difference with resepect to the current defination.

**Comment by Abhishekc95** [comment]
another question is , you are multiplying 1e-6 to pde loss. its almost like we are treating it as negligible. but again the solution is matching closely with fem. can you please explain about this?

**Comment by hl4220** [comment]
This is to keep the value of pde loss to be in a similar level with the boundary condition loss. You can have a try: if there is no 1e-6, the pde would be too large compared to the bc loss.

**Comment by Abhishekc95** [comment]
I have couple more questions if you dont mind.
1) In getting the hybrid eigen vectors from fem for example 2 problem. can it be done analytically ? or is using FEM necessary even for simple square or rectangle domains?
2) in example 2, using fem do we have to solve the unconstrained eigen value problem for K and then add those with constrained Ku (as you have done in matlab code (e = e +eu) ) for any general problem?.
3) you have scaled both e and eu in the code with 10, is this done to avoid vanishing gradients in neural network?
4) in example 2, the eigen vectors dont satisfy the bc's implicitly, you have used soft constraints, did you compare this with traditional PINN with hard bc without using LBO's? 

edit: my questions are related to solution for strong form in example 2

**Comment by hl4220** [comment]
1. yes it can be done analytically. Actually an anlytical form is even better, if you can find it;
2. In this problem yes;
3. It doesn't matter, I just want to normalise those fields, or it would be too small;
4. No I haven't done those comparison. example 2 is served as a benchmark for quick start of the model.

**Comment by Abhishekc95** [comment]
Hey, I would like to implement the wave propagation problem in 2d domain. A simple plane wave similar to your 1D problem. I can send you an email, can you help me out?

**Comment by hl4220** [comment]
Yes please send me by email: haolin.li20@imperial.ac.uk
